### PR TITLE
Fix worker and multiplex workers for DexBuilder and Desugar actions

### PR DESCRIPTION
**Pull request by Bencodes**
Fixing up the DexBuilder and Desugar actions so that they correctly spawn worker or multiplexed worker actions. `--modify_execution_info` doesn't work as expected and is not additive, which results in the previous execution infos being removed.

### CODE REVIEW

**Comment by Bencodes** [comment]
Opened up a cherry pick here for when this lands in master https://github.com/bazelbuild/bazel/pull/17965

**Review by ted-xie — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/rules/android/AndroidConfiguration.java [hunk 1]
```diff
@@ -533,14 +533,6 @@ public static class Options extends FragmentOptions {
         help = "dx flags supported in tool that groups classes for inclusion in final .dex files.")
     public List<String> dexoptsSupportedInDexSharder;
 
-    @Option(
-        name = "use_workers_with_dexbuilder",
```
[the remark this answers is not in the document]
Comment (Bencodes) [reply src/main/java/com/google/devtools/build/lib/rules/android/AndroidConfiguration.java:537]: Updated

**Review by ted-xie — APPROVED** [review]
Overall LGTM, there is just one more thing to fix (that we can handle internally) - we'll have to move the deprecated @Option stanza from BazelBuildGraveyardOptions class to the BuildGraveyardOptions class.

**Comment by keertk** [comment]
@bazel-io fork 6.2.0

**Pull request closed without merging**